### ISSUE TITLE
Training loss of LenPredictor.

### ISSUE THREAD

**Issue by huangf79** [gallilmaimon/DISSC#4]
Hey, I open this new issue as you suggested. For anyone interested, I raised new question [here](https://github.com/gallilmaimon/DISSC/issues/3#issuecomment-2115583575),

> Hey. Sorry to reopen this issue, I tried to train a LenPredictor but got bad results. Even after 200 epochs, the LenSumLoss was still more than 1800 and the predicted durations were unreasonable. (1) I wonder what the normal loss value is, also that for PitchPredictor. (I used just part of your code, maybe it's caused by my wrong implementation) (2) BTW, does the mean and var of the dataset matters? I set them 0 and 1 for convenience.

and the author gave the answer [here](https://github.com/gallilmaimon/DISSC/issues/3#issuecomment-2115871345).

> Hey again, generally speaking I prefer if you open a new issue for each separate thing so that people in the future can find it more easily :)
> 
> About your questions: (1) a normal loss value depends on the distribution of the dataset (specifically the duration distribution), a general rule of thumb is that it should be a noticeably better loss than simply guessing the average length or the common value or something. Are you using one of the described datasets or a custom one? I would look at the distribution of gt length to see if maybe there is bug or maybe the Hubert model doesn't output good tokens for your dataset. Anyway your loss sounds like some sort of bug. (2) the mean and var of which dataset? Of the durations? This can impact the absolute loss scale which would impact learning rate and also the relative weight to big errors (because of the MSE). This won't necessarily cause a big difference but hard to guarantee.

**Comment by huangf79** [comment]
Here are my new comments: 
(1) That makes sense. I trained it on ESD, and used HuBERT features from layer 9 with k=500 (publicly available). I will train it with features l6k100 (the same as the paper) tonight and see how it goes. It would be very helpful if you could provide me with a normal loss.
(2) The mean and var of [training set](https://github.com/gallilmaimon/DISSC/blob/main/train_len_predictor.py#L29), (durations of ESD in my case). Anyway, I get it. 

Sorry for missing some details in my question,  I was quite confused at that time.

**Comment by gallilmaimon** [comment]
I am afraid I can't remember of the top of my head the exact loss value. As the loss is an average square of the length error per sample (+a small other term) it means that the average duration prediction error in your case is ~40 which is far from what I observed. As a quick test you could use the pretrained model provided [here](https://drive.google.com/drive/u/1/folders/1ZNT9cyK5CPwFQzu6WY2c4wS56fNQf57a) (the same one as linked in the repo), based on data extracted exactly like this repository and get the expected loss value. As another simple sanity check you could calculate the mean of the dataset and see what returning that gives you for an overall MSE, results of the trained model should be (notably) better than this. If I really try to remember I think the loss was in the range of 1-10, but best to test it again.

The normalisation could also impact the MSE scale if you calculated it on the normalised values. For instance the normalisation could make the data have many very negative values because the length distribution does not follow a normal distribution, but rather a decaying distribution with most values at 1, then less at 2, etc. but with also some outliers with length ~40.

I hope these can provide some help and insight, let me know if you manage to find the issue or have problems recreating any of the results with our full code on our datasets.

**Comment by huangf79** [comment]
Thank you for your patience. I will try.

**Comment by gallilmaimon** [comment]
Hello, hope you were successful in your attempt.

If you need anymore help please let me know, otherwise I will close this issue as stale :)
